packages/flutter/lib/src/material/dialog.dart

```diff
@@ -481,12 +486,15 @@ class _DialogRoute<T> extends PopupRoute<T> {
 Future<T> showDialog<T>({
   @required BuildContext context,
   bool barrierDismissible: true,
-  @required Widget child,
+  @deprecated Widget child,
+  @required WidgetBuilder builder,
 }) {
   return Navigator.of(context, rootNavigator: true).push(new _DialogRoute<T>(
```

Comment:
assert that one of child and builder is non-null and that the other is null.

Reply:
Done